autoload/airline/extensions/branch.vim

```diff
@@ -68,37 +71,35 @@ function! s:get_git_branch(path)
 endfunction
 
 function! s:get_git_untracked(file)
```

Comment:
There's a bit of code duplication with respect to hg and git. Notice that you have performed the async path in a VCS-independent fashion, but the synchronous paths are still separate. As a future TODO you might consider removing this duplication. I may do it if you are OK with this kind change.